### PR TITLE
Fix issue #4844

### PR DESCRIPTION
**Pull request by chinmay81098**
Automatic copy of the selected code snippet to clipboard feature added

### CODE REVIEW

**Comment by vigsterkr** [comment]
yey works!
https://www.shogun.ml/cookbook_pr/99d6ad3b3e5f35805431f38570ba4c3d6aaa14a6/html/examples/binary/linear_support_vector_machine.html

**Comment by geektoni** [comment]
Shouldn't we add some sort of "Copied!" notification to show the user what happened? :/

**Comment by gf712** [comment]
> Shouldn't we add some sort of "Copied!" notification to show the user what happened? :/

I think so too! It took me a while to understand what was happening :D But thank you for adding this feature already!

**Comment by chinmay81098** [comment]
> Shouldn't we add some sort of "Copied!" notification to show the user what happened? :/

Yes, thanks for the feedback. Its added now.

**Comment by gf712** [comment]
@chinmay81098 how difficult would it to add something like [this](https://i.stack.imgur.com/3GBr2.png) rather than an alert?

**Comment by chinmay81098** [comment]
> @chinmay81098 how difficult would it to add something like [this](https://i.stack.imgur.com/3GBr2.png) rather than an alert?

A bit, but I have tried to implement something similar. Kindly review.

**Comment by geektoni** [comment]
This makes it more clear @chinmay81098 ! However, I am still thinking that maybe we would need something more "visible" since the user may not know that by selecting the code it will be also copied. 

Could it be possible to add some kind of button on the top right corner of the code snippets that copies everything in the snippet inside the clipboard (like [this](https://user-images.githubusercontent.com/3296552/77080192-4b573100-69f9-11ea-9ded-3a9e6f717d90.png))?

**Comment by gf712** [comment]
> This makes it more clear @chinmay81098 ! However, I am still thinking that maybe we would need something more "visible" since the user may not know that by selecting the code it will be also copied.
> 
> Could it be possible to add some kind of button on the top right corner of the code snippets that copies everything in the snippet inside the clipboard (like [this](https://user-images.githubusercontent.com/3296552/77080192-4b573100-69f9-11ea-9ded-3a9e6f717d90.png))?

I think what @chinmay81098 seems to be pretty good already. That would be even nicer though. Do you think we could have both or would it be too much?

File: doc/cookbook/source/templates/page.html [hunk 1]
```diff
@@ -20,6 +20,25 @@
...
+	    }    
+      var selected = this;
+      timeout = setTimeout(function() {
+      $(selected).tooltip("hide");
+      }, 1000);
+    });
+
+    $('.highlight').tooltip({
+	    trigger:'manual',
+      placement:'bottom',
```
Comment (gf712) [inline doc/cookbook/source/templates/page.html:39]: can the tooltip be placed relative to the cursor? I am thinking it is a bit arbitrary that it is in the middle

**Comment by karlnapf** [comment]
In the link above, how to make it work? :D sorry I am confused

**Comment by vigsterkr** [comment]
https://www.shogun.ml/cookbook_pr/9546d036795c57e7d1fdf09dc13a14762df970fa/html/examples/binary/linear_support_vector_machine.html

commit hash

**Comment by vigsterkr** [comment]
to make it work, just select text in code section

**Comment by karlnapf** [comment]
Ah yes!

One thing I would like to suggest is also to have a button to view/copy the *whole* code listing (as in, including the bits that are not rendered)

Since we are touching this @chinmay81098 you wanna go ahead with that?

**Comment by vigsterkr** [comment]
@karlnapf could we have that in another pr? :) that's quite some more hacking :)

**Comment by karlnapf** [comment]
yes of course sorry :)
I think there is an issue for that already (although it might have been closed as stale)

**Comment by karlnapf** [comment]
It is #4020

File: doc/cookbook/source/templates/page.html (hunk 1 above)
Comment (chinmay81098) [reply doc/cookbook/source/templates/page.html:39]: Yes, but for that I need to wrap the text in one single element and then add tooltip to it,right now each letter has its own tag that's why I had to target the complete box. Do I need to look into sgexample.py for this?

**Comment by chinmay81098** [comment]
> > This makes it more clear @chinmay81098 ! However, I am still thinking that maybe we would need something more "visible" since the user may not know that by selecting the code it will be also copied.
> > Could it be possible to add some kind of button on the top right corner of the code snippets that copies everything in the snippet inside the clipboard (like [this](https://user-images.githubusercontent.com/3296552/77080192-4b573100-69f9-11ea-9ded-3a9e6f717d90.png))?
> 
> I think what @chinmay81098 seems to be pretty good already. That would be even nicer though. Do you think we could have both or would it be too much?

Both will be nice to have, but @karlnapf mentioned that there is already a pr for adding a button so I'll just follow that issue there and contribute if required.

**Pull request merged**